packages/workflows/src/definitions/store/carts/add-shipping-method.ts

```diff
@@ -0,0 +1,257 @@
+import { validateShippingOptionForCart } from "../../../handlers/store/carts/validate-shipping-option-for-cart"
+import { exportWorkflow, pipe } from "../../../helper"
+
+export type AddShippingMethodWorkflowInputData = {
+  cart: any // Cart
+}
+// & StorePostCartsCartShippingMethodReq
+
+export enum AddShippingMethodWorkflowActions {
+  prepare = "prepare",
+  validateFulfillmentData = "validateFulfillmentData",
+  validateLineItemShipping = "validateLineItemShipping",
+  getOptionPrice = "getOptionPrice",
+  createShippingMethods = "createShippingMethods",
+  cleanUpShippingMethods = "cleanUpShippingMethods",
+  adjustFreeShipping = "adjustFreeShipping",
+  cleanUpPaymentSessions = "cleanUpPaymentSessions",
+  updatePaymentSessions = "updatePaymentSessions",
+  result = "result",
+}
+
+export const addShippingMethodWorkflowSteps: TransactionStepsDefinition = {
+  next: {
+    // retrieve cart + custom shipping options
+    action: AddShippingMethodWorkflowActions.prepare,
+    noCompensation: true,
+    saveResponse: true,
+    next: [
+      {
+        // validate fulfillment data
+        action: AddShippingMethodWorkflowActions.validateFulfillmentData,
+        noCompensation: true,
+        saveResponse: true,
+      },
+      {
+        // validate line item shipping
+        action: AddShippingMethodWorkflowActions.validateLineItemShipping,
+        noCompensation: true,
+        next: {
+          // get price of shipping option
+          action: AddShippingMethodWorkflowActions.getOptionPrice,
+          noCompensation: true,
+          saveResponse: true,
+          next: {
+            // create the shipping method
+            action: AddShippingMethodWorkflowActions.createShippingMethods,
+            noCompensation: true,
+            next: {
+              // delete other shipping methods with same profile id
+              action: AddShippingMethodWorkflowActions.cleanUpShippingMethods,
+              next: {
+                // adjust free shipping discount wrt new shipping method
+                action: AddShippingMethodWorkflowActions.adjustFreeShipping,
+                next: {
+                  // clean up payment sessions
+                  action:
+                    AddShippingMethodWorkflowActions.cleanUpPaymentSessions,
+                  next: {
+                    // update the payment sessions on the cart
+                    action:
+                      AddShippingMethodWorkflowActions.updatePaymentSessions,
+                    // retrieve cart with totals
+                    next: {
+                      action: AddShippingMethodWorkflowActions.result,
+                      noCompensation: true,
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    ],
+  },
+}
+
+const handlers = new Map([
+  [
+    AddShippingMethodWorkflowActions.prepare,
+    {
+      invoke: prepareAddShippingMethodToCartWorkflowData,
```

Comment:
discussion: If you do it that way, you won't have anything in the data, you will have to access the payload which means that the handler is only usable with a payload object. I suggest that you grab and set the data from the pipe